plugins/quickwit.php

```diff
@@ -0,0 +1,287 @@
+        $this->extractRecursive($data, $this->resultsMapping, $results);
+        return $this->sortResults($results);
+    }
+
+    private function sortResults($results): array
+    {
+        return array_map(
+            function ($row) {
+                ksort($row, SORT_STRING);
+                return $row;
+            }, $results
+        );
+    }
+
+    private function extractRecursive(
+        $data,
+        $structure,
+        &$results,
+        &$currentResult = null
+    ): void {
+        if ($currentResult === null) {
+            $currentResult = [];
+        }
+
+        foreach ($structure as $key => $value) {
+            if ($key === "[]") {
+                if (is_array($data)) {
+                    foreach ($data as $item) {
+                        $subResult = [];
+                        $this->extractRecursive($item, $value, $results,
+                            $subResult);
+                        $results[] = $subResult;
+                    }
+                }
+            } elseif ($key === "single_value") {
+                $subResult = [];
+                $this->extractRecursive(
+                    $data, $value, $results, $subResult
+                );
+                $results[] = $subResult;
+            } elseif (is_array($value)) {
+                if (array_key_exists($key, $data)) {
+                    $this->extractRecursive(
+                        $data[$key], $value, $results, $currentResult
+                    );
+                }
+            } else {
+                if (array_key_exists($key, $data)) {
+                    $fieldInfo = explode(':', $value);
+                    $fieldName = trim($fieldInfo[0]);
+                    $fieldType = isset($fieldInfo[1]) ? trim($fieldInfo[1])
+                        : null;
+
+                    $fieldValue = $data[$key];
+                    if ($fieldType === 'float') {
+                        $currentResult[$fieldName] = round((float) $fieldValue,
+                            4);
+                    } elseif ($fieldType === 'int') {
+                        $currentResult[$fieldName] = (int) $fieldValue;
+                    } else {
+                        $currentResult[$fieldName] = $fieldValue;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param array<string,mixed> $results
+     *
+     * @return array<string,mixed>
+     */
+    protected function filterResults(
+        array $results,
+        $elasticStyle = false
+    ): array {
+        $filtered = [];
+        foreach ($results as $row) {
+            $ar = [];
+
+            if ($elasticStyle) {
+                $row = $row['_source'];
+            }
+
+            foreach ($row as $k => $v) {
+                if ($k === 'id') {
+                    continue;
+                } // removing id from the output sice Elasticsearch can't return it https://github.com/elastic/elasticsearch/issues/30266
```

Comment:
```suggestion
                } // removing id from the output since Elasticsearch can't return it https://github.com/elastic/elasticsearch/issues/30266
```